### PR TITLE
chores: flashcards list

### PR DESCRIPTION
**Pull request by SzDabrowski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Create UI for displaying multiple cards according to the design.

### Requirements
1. Header of the app should be fixed.
2. Cards should be displayed in the vertical alignment with spacing 12px.
3. Cards list should scroll.
4. Only UI should be created in this task scope.

### Code Changes Review

#### package.json
The addition of "node": "^20.5.0" is not related to the task requirements and seems unnecessary.

#### src/App.css
The changes in this file meet the requirements. The `.cardList_container` class is set to display its children (the cards) in a vertical alignment with a margin-top of 12px, which will create the required spacing between cards.

#### src/App.tsx
The changes in this file meet the requirements. The `AppHeader` is fixed at the top of the `AppLayout`. The `cardList` state is used to store the cards and they are displayed in a `div` with the class `cardList_container`, which will make the card list scrollable. The `editMode` state and `changeMode` function seem to be related to functionality beyond displaying the UI, which is out of scope for this task.

#### src/components/AppHeader.tsx
No significant changes were made in this file related to the task requirements.

#### src/components/AppLayout.module.css
The changes in this file meet the requirements. The `AppLayout` is set to display its children (the header and card list) in a column, which will make the header fixed at the top.

#### src/components/NewCard.module.css
This file was renamed but no changes were made in it related to the task requirements.

#### src/components/NewCard.tsx
This file was renamed and some changes were made in it. The `Card` component was renamed to `NewCard`. The `savePageClick` function was added, which seems to be related to functionality beyond displaying the UI, which is out of scope for this task.

### Conclusion
The code changes mostly meet the task requirements. However, there are some changes that seem unnecessary or out of scope for this task, such as the addition of "node": "^20.5.0" in package.json and the `editMode` state and `changeMode` function in src/App.tsx.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Create UI for displaying multiple cards according to the design.

### Requirements
1. Header of the app should be fixed.
2. Cards should be displayed in the vertical alignment with spacing 12px.
3. Cards list should scroll.
4. Only UI should be created in this task scope.

### Code changes review

#### src/App.css
The changes in this file seem to be in line with the requirements. The `.cardList_container` class has been added to handle the card list display and scrolling.

#### src/App.tsx
The changes in this file seem to cover the requirements. The `NewCard` component is used instead of `Card`, and the state for `editMode` and `cardList` has been added. The `changeMode` function has been added to switch between edit and view modes, and the `addCard` function has been added to add new cards to the list. The `cardList` is displayed inside a div with the class `cardList_container`, which should ensure that the cards are displayed in a vertical alignment with scrolling.

#### src/components/AppHeader.tsx
No significant changes have been made in this file, so it's hard to comment on whether it meets the requirements or not.

#### src/components/AppLayout.module.css
The changes in this file seem to be in line with the requirements. The `display: flex;` and `flex-direction: column;` properties have been added to ensure that the header is fixed and the cards are displayed in a vertical alignment.

#### src/components/NewCard.module.css
This file has been renamed, so it's hard to comment on whether it meets the requirements or not without seeing the actual content of the file.

#### src/components/NewCard.tsx
This file has been renamed from `Card.tsx` to `NewCard.tsx`, and some changes have been made. The `NewCard` component now uses the `setEditMode` prop to switch between edit and view modes, which seems to be in line with the requirements. The `savePageClick` function has been added to handle saving of the card.

In conclusion, based on the provided code changes, it seems that they cover the task requirements. However, without seeing the actual content of the renamed files (`NewCard.module.css` and `NewCard.tsx`), it's hard to make a definitive conclusion.

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
-import {Card} from "./components/Card";
+import {NewCard} from "./components/NewCard";
+import React, { useState } from "react";
 
 import "./App.css";
-import React from "react";
 
 function App() {
+  const [editMode, setEditMode] = useState<boolean>(true);
+  const [cardList, setCardList] = useState<JSX.Element[]>([]);
```
Comment (Sumick) [inline src/App.tsx:10]: **issue:** Keeping array of elements is not react way of doing it. You should handle data structure here, as array of objects, then you should map this data as react components. You will find a good example in react docs of how can you render lists: https://react.dev/learn/rendering-lists

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
 function App() {
+  const [editMode, setEditMode] = useState<boolean>(true);
+  const [cardList, setCardList] = useState<JSX.Element[]>([]); 
+
+  const changeMode = () => {
+    setEditMode(true);
+  };
+
+  const addCard = () => {
+    const newCard = <NewCard editMode={editMode} setEditMode={setEditMode} />;
```
Comment (Sumick) [inline src/App.tsx:17]: **issue:** Adding new card should set data object instead of JSX element. However it was not in this task scope, so you can pass this for now

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+    setCardList([...cardList, newCard]); // Dodajemy nowy komponent Card do listy
+    console.log("added");
+  };
+
   return (
     <AppLayout>
-      <AppHeader cardsNumber = {1}/>
+      <AppHeader CurrentCardsNumber={cardList.length} changeMode={addCard} />
```
Comment (Sumick) [inline src/App.tsx:24]: **nitpick:** prop names should be written with `camelCase` instead of `PascalCase`

File: src/App.tsx [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+      <div className="cardList_container">
+        {editMode ? cardList : null}
```
Comment (Sumick) [inline src/App.tsx:26]: **issue:** You don't have any control over rendered elements now. We should render JSX elements inline here with `map` function

**Review by Sumick — CHANGES_REQUESTED** [review]
There is an issue of how you render card elements, as it's completed non react way - check out React docs for example: https://react.dev/learn/rendering-lists. Most of my comments are related to this problem.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review

## Task Requirements Validation

1. **Header of the app should be fixed** - The code changes do not explicitly show that the header is fixed. However, the `AppLayout` component has been modified to use flexbox which could potentially make the header fixed. This requirement is not clearly met in the provided code changes.

2. **Cards should be displayed in the vertical alignment with spacing 12px** - The `cardList_container` class in `App.css` has been modified to use flexbox with a column direction, which would align the cards vertically. Additionally, a margin-top of 12px has been added to all direct children of `cardList_container`, which would create a spacing of 12px between the cards. This requirement is met.

3. **Cards list should scroll** - The `cardList_container` class in `App.css` has been given an `overflow-y: auto;` property, which would allow the cards list to scroll if the content exceeds the container's height. This requirement is met.

4. **Only UI should be created in this task scope** - The code changes include some logic for handling state and events, such as adding a new card and switching between edit and view modes. This goes beyond just creating UI and does not meet this requirement.

## Bugs and Issues

- In `src/App.tsx`, the `addCard` function creates a new `NewCard` component but does not do anything with it. This will not add a new card to the UI.

- In `src/components/Card.tsx`, the `changePageClick` function toggles the `nextPage` state but does not update the displayed text accordingly.

- In `src/components/NewCard.tsx`, the `savePageClick` function sets `editMode` to false but does not save the changes made to the card.

- In `src/components/Card.module.css`, there are some unused styles left over from the previous version of the file. These should be removed to avoid confusion.

- In `src/components/AppHeader.tsx`, the prop name `currentCardsNumber` does not follow the camelCase convention for JavaScript. It should be `currentCardNumber`.

## Summary

The code changes partially meet the task requirements. The layout and scrolling of the cards list have been implemented correctly, but there are issues with the fixed header and the scope of the task. Additionally, there are several bugs and issues that need to be addressed.

**Review by Sumick — APPROVED** [review]
Looks good now! Great job 🙌 However Im not sure why do you have so many changes in this task scope? I mean `NewCard` related logic. Looks like it was main reason why Nerdy couldn't let you pass
